portability-core/src/main/java/org/dataportabilityproject/cloud/google/GooglePersistentKeyValueStore.java

```diff
@@ -185,7 +208,11 @@ private Entity createEntity(String key, Map<String, Object> data) throws IOExcep
     return builder.build();
   }
 
-  private Key getKey(String key) {
-    return datastore.newKeyFactory().setKind(KIND).newKey(key);
+  private Key getKey(String jobId) {
+    return datastore.newKeyFactory().setKind(KIND).newKey(jobId);
+  }
+
+  private JobState getJobState(Entity entity) {
+    return JobState.valueOf(entity.getString(PortabilityJobConverter.JOB_STATE));
```

Comment:
Should we check that 'entity.getString(PortabilityJobConverter.JOB_STATE))' is not null?